inspire_mitmproxy/services/management_service.py

```diff
@@ -108,17 +114,25 @@ def put_config(self, request: MITMRequest):
         try:
             config_update = json_loads(request.body)
             self.config.update(config_update)
-            self.config_propagate()
+            self.options_propagate()
         except (JSONDecodeError, ValueError):
             raise InvalidRequest(self.name, request)
 
     def post_config(self, request: MITMRequest):
         try:
             self.config = json_loads(request.body)
-            self.config_propagate()
+            self.options_propagate()
         except JSONDecodeError:
             raise InvalidRequest(self.name, request)
 
+    def set_recording(self, request: MITMRequest):
+        try:
+            recording_opts = json_loads(request.body)
+            self.is_recording = recording_opts['enable']
+            self.options_propagate()
+        except (JSONDecodeError, KeyError, TypeError):
+            raise InvalidRequest(self.name, request)
```

Comment:
This is returning 'None', that in the call ends up being 'null' as response. Can we do a bit better and return something nicer like:
```
{
    "status": "ok",
    "message": "Recording enabled".
}
```
or similar?

I know that the status itself is already in the response status_code, but even then, the message helps when developing/debugging.

Maybe we can also generalize that formatted response :/

Reply:
Hmm... I think we can do better, for example, returning the new state of the object that is updated. For example when you update the config (POST or PUT to it), the response can be the new config status. And when you post to the /record endpoint, the response should be the record status:
```
{
    "enabled": true,
}
```
For example, and we should add that status show when the request is a GET :), I'd like that (probably a different PR/task)

Reply:
Made a ticket for doing it globally, https://its.cern.ch/jira/browse/INSPIR-843, will go with the something simple now